packages/client/hmi-client/src/model-representation/mira/mira.ts

```diff
@@ -0,0 +1,323 @@
+import _ from 'lodash';
+import { IGraph } from '@graph-scaffolder/types';
+import {
+	extractSubjectControllersMatrix,
+	extractSubjectOutcomeMatrix,
+	removeModifiers
+} from './mira-util';
+import type { MiraModel, MiraTemplateParams, TemplateSummary } from './mira-common';
+
+/**
+ * Collection of MMT related functions
+ * */
+export const getContextKeys = (miraModel: MiraModel) => {
+	const modifierKeys = new Set<string>();
+
+	miraModel.templates.forEach((t) => {
+		if (t.subject && t.subject.context) {
+			Object.keys(t.subject.context).forEach((key) => modifierKeys.add(key));
+		}
+		if (t.outcome && t.outcome.context) {
+			Object.keys(t.outcome.context).forEach((key) => modifierKeys.add(key));
+		}
+		if (t.controller && t.controller.context) {
+			Object.keys(t.controller.context).forEach((key) => modifierKeys.add(key));
+		}
```

Comment:
could shorten these to `t.subject?.context`

Reply:
sure